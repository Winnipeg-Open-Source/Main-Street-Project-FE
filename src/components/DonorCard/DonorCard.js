import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Card, Text } from 'pcln-design-system';

const StyledCard = styled(Card)`
    border-radius: 4px;
`;

function DonorCard ({ className, name, lastDonationDate }) {
    return (
        <StyledCard className={className} color='background.lightest' width={1} p={2}>
            <Text color='text.darkest' fontSize={1} bold>{name}</Text>
            <Text color='text.light' fontSize={1}>Last Donation: {lastDonationDate || '-'}</Text>
        </StyledCard>
    );
}

DonorCard.displayName = 'DonorCard';

DonorCard.propTypes = {
    className: PropTypes.string,
    name: PropTypes.string,
    lastDonationDate: PropTypes.string,
};

DonorCard.defaultProps = {
    className: '',
};

export default memo(DonorCard);
