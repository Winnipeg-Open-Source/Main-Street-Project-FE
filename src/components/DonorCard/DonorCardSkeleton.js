import React, { memo } from 'react';
import PropTypes from 'prop-types';
import Card from 'components/Card';
import Skeleton from 'components/Skeleton';

function DonorCardSkeleton ({ className }) {
    return (
        <Card className={className}>
            <Skeleton width='20%' height={14} />
            <Skeleton width='40%' height={14} />
        </Card>
    );
}

DonorCardSkeleton.displayName = 'DonorCardSkeleton';

DonorCardSkeleton.propTypes = {
    className: PropTypes.string,
};

DonorCardSkeleton.defaultProps = {
    className: '',
};

export default memo(DonorCardSkeleton);
