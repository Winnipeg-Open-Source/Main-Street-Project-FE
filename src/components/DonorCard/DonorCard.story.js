import React from 'react';
import DonorCard from './DonorCard';

const props = {
    name: 'Costco',
    lastDonationDate: '2019 December 20',
};

export const HasNotDonated = () => <DonorCard name='Costco' />;

export const HasDonated = () => <DonorCard {...props} />;

export default {
    title: 'DonorCard',
    component: DonorCard,
};
