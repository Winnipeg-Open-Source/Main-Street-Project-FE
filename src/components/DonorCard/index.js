import WithSkeleton from 'hocs/WithSkeleton';
import DonorCard from './DonorCard';
import DonorCardSkeleton from './DonorCardSkeleton';

export default WithSkeleton(DonorCardSkeleton, DonorCard);
