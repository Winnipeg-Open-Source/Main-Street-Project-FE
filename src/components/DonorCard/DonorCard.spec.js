import React from 'react';
import { fireEvent } from '@testing-library/react';
import renderWithTheme from 'tests/helpers/renderWithTheme';
import DonorCard from './DonorCard';
import DonorCardSkeleton from './DonorCardSkeleton';

const props = {
    name: 'Costco',
    lastDonationDate: '2019 December 20',
};

describe ('DonorCard', () => {
    it ('renders without crashing', () => {
        const { asFragment } = renderWithTheme(<DonorCard />);

        expect(asFragment()).toMatchSnapshot();
    });

    it ('skeleton renders without crashing', () => {
        const { asFragment } = renderWithTheme(<DonorCardSkeleton />);

        expect(asFragment()).toMatchSnapshot();
    });

    it ('has not donated', async () => {
        const { getByText } = renderWithTheme(<DonorCard name='Costco' />);

        const name = await getByText('Costco');
        const lastDonation = await getByText('Last Donation: -');

        expect(name.textContent).toBe('Costco');
        expect(lastDonation.textContent).toBe('Last Donation: -');
    });

    it ('has donated', async () => {
        const { getByText } = renderWithTheme(<DonorCard {...props} />);

        const name = await getByText('Costco');
        const lastDonation = await getByText('Last Donation: 2019 December 20');

        expect(name.textContent).toBe('Costco');
        expect(lastDonation.textContent).toBe('Last Donation: 2019 December 20');
    });

    it ('is clickable', () => {
        const mockClick = jest.fn();
        const { getByTestId } = renderWithTheme(<DonorCard onClick={mockClick} />);

        const Card = getByTestId('donor-card');
        expect(mockClick).toHaveBeenCalledTimes(0);
        fireEvent.click(Card);
        expect(mockClick).toHaveBeenCalledTimes(1);
    });
});
