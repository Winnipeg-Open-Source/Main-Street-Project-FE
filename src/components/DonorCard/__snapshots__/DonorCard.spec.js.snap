// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonorCard renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-qOvoE hIfWnJ sc-qWSGh dpgLb"
  >
    <div
      class="sc-AxirZ sc-pQfvp sc-pAMRk isCbCu"
      color="background.lightest"
      data-testid="donor-card"
      width="1"
    >
      <div
        class="sc-AxhUy gpDHNu"
        color="text.darkest"
        font-size="1"
      />
      <div
        class="sc-AxhUy enVSKo"
        color="text.light"
        font-size="1"
      >
        Last Donation: -
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DonorCard skeleton renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-qOvoE hIfWnJ sc-qWSGh dpgLb"
  >
    <div
      class="sc-AxirZ sc-pQfvp sc-pAMRk djnmSu sc-pRFZy fVRgPn"
      color="background.lightest"
      width="1"
    >
      <span>
        <span
          class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
          style="width: 25%; height: 14px;"
        >
          ‌
        </span>
      </span>
      <span>
        <span
          class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
          style="width: 50%; height: 14px;"
        >
          ‌
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
